File: src/IotBbq.App/IotBbq.JsApp/src/app/services/BbqEvent.ts
export interface IBbqEvent {
  id: string;
  name: string;
  eventDate: Date;
  turnInTime: Date;
}

export class BbqEvent implements IBbqEvent {
  id: string;
  name: string;
  eventDate: Date;
  turnInTime: Date;

  public load(other: IBbqEvent) {
    this.id = other.id;
    this.name = other.name;
    this.eventDate = other.eventDate;
    this.turnInTime = other.turnInTime;
  }

  public writeTo(other: IBbqEvent) {
    other.id = this.id;
    other.name = this.name;
    other.eventDate = this.eventDate;
    other.turnInTime = this.turnInTime;
  }
}

File: src/IotBbq.App/IotBbq.Web.Model/BbqEvent.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace IotBbq.Web.Model
{
    public class BbqEvent : EntityBase
    {
        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("eventDate")]
        public DateTime EventDate { get; set; }

        [JsonProperty("turnInTime")]
        public DateTime TurnInTime { get; set; }

        public List<BbqItem> Items { get; set; }

        public List<SmokerLog> SmokerLogs { get; set; }
    }
}
